item-review-viewer/src/main/java/org/smarterbalanced/itemreviewviewer/web/services/GitLabUtils.java

```diff
@@ -30,6 +48,24 @@ public static String makeItemId(String bankKey, String itemKey){
         return itemId;
     }
 
+    //forms a stim id.
+    public static String makeStimId(String bankKey, String itemKey){
+        return "stim-" + bankKey + "-" + itemKey;
+    }
+
+    //takes a valid item id and makes a qualified item ID
+    public static String makeQualifiedItemId(String itemId, String version){
+        itemId = itemId.toLowerCase();
+        if(itemId.contains("item-")){
+            itemId = itemId.replace("item-",  "");
+        }
+        String qualifiedItemId = "i-" + itemId;
+        if(version != null && StringUtils.isNotEmpty(version)){
+            qualifiedItemId.concat("-" + version);
```

Comment:
Do you mean `return qualifiedItemId.concat("-" + version);` ?

Reply:
No because version can be null if we just return that then we could end up with an Id that looks like "i-187-3000-"

Reply:
We only want to add the last - if version is not null or not empty.

Reply:
`concat` function returns the string concatenated with "-" and version variable doesn't change, so we should use the return value from `concat` function.

Reply:
I changed it to qualifiedItemId = qualifiedItemId + "-" + version.